caravel/models.py

```diff
@@ -1059,8 +1240,78 @@ def fetch_metadata(self):
         if not self.main_dttm_col:
             self.main_dttm_col = any_date_col
 
+    def alter_json_metadata(self, **kwargs):
+        d = self.dict_metadata
+        d.update(kwargs)
+        self.json_metadata = json.dumps(d)
+
+    @property
+    def dict_metadata(self):
+        if self.json_metadata:
+            return json.loads(self.json_metadata)
+        return {}
+
+    @classmethod
+    def import_obj(cls, datasource_to_import, import_time=None):
```

Comment:
There's a lot in common in all these `import_obj` methods. It'd be nice to distill some of it into re-usable chunks, maybe adding a CaravelModel base class and adding a get_or_create @classmethod. The second answer here is inspiring:
http://stackoverflow.com/questions/2546207/does-sqlalchemy-have-an-equivalent-of-djangos-get-or-create

And maybe a generic `import_obj` if that's reasonable though that might be tricky with relationships, so not a requirement for this PR...